### PR TITLE
msg/async/rdma: automatically set RDMAV_HUGEPAGES_SAFE according to conf

### PR DESCRIPTION
**Pull request by Adirl**
Signed-off-by: DanielBar-On <danielbo@mellanox.com>

### CODE REVIEW

**Comment by Adirl** [comment]
@yuyuyu101 
please take a look

File: src/msg/async/rdma/RDMAStack.cc [hunk 1]
```diff
@@ -516,12 +516,23 @@ RDMAStack::RDMAStack(CephContext *cct, const string &t): NetworkStack(cct, t)
...
   //
+
   int rc = ibv_fork_init();
   if (rc) {
      lderr(cct) << __func__ << " failed to call ibv_for_init(). On RDMA must be called before fork. Application aborts." << dendl;
      ceph_abort();
   }
 
+  ldout(cct, 20) << __func__ << " ms_async_rdma_enable_hugepage value is: " << cct->_conf->ms_async_rdma_enable_hugepage <<  dendl;
+  if (cct->_conf->ms_async_rdma_enable_hugepage){
```
Comment (yuyuyu101) [inline src/msg/async/rdma/RDMAStack.cc]: blank before bracket

File: src/msg/async/rdma/RDMAStack.cc [hunk 2]
```diff
@@ -548,6 +559,10 @@ RDMAStack::RDMAStack(CephContext *cct, const string &t): NetworkStack(cct, t)
 
 RDMAStack::~RDMAStack()
 {
+  if (cct->_conf->ms_async_rdma_enable_hugepage){
```
Comment (yuyuyu101) [inline src/msg/async/rdma/RDMAStack.cc]: ditto

File: src/msg/async/rdma/RDMAStack.cc [hunk 3]
```diff
@@ -516,12 +516,23 @@ RDMAStack::RDMAStack(CephContext *cct, const string &t): NetworkStack(cct, t)
...
   int rc = ibv_fork_init();
   if (rc) {
      lderr(cct) << __func__ << " failed to call ibv_for_init(). On RDMA must be called before fork. Application aborts." << dendl;
      ceph_abort();
   }
 
+  ldout(cct, 20) << __func__ << " ms_async_rdma_enable_hugepage value is: " << cct->_conf->ms_async_rdma_enable_hugepage <<  dendl;
+  if (cct->_conf->ms_async_rdma_enable_hugepage){
+    rc =  setenv("RDMAV_HUGEPAGES_SAFE","1",1);
+    ldout(cct, 20) << __func__ << " RDMAV_HUGEPAGES_SAFE is set as: " << getenv("RDMAV_HUGEPAGES_SAFE") <<  dendl;
```
Comment (yuyuyu101) [inline src/msg/async/rdma/RDMAStack.cc]: ldout(cct, 1) should be better

**Comment by Adirl** [comment]
@yuyuyu101 
Done

**Comment by Adirl** [comment]
@yuyuyu101 @tchaikov 
Ping

**Pull request merged**